src/Plugin/views/argument_default/Membership.php

```diff
@@ -0,0 +1,127 @@
+
+/**
+ * Default argument plugin to provide the group from the current context.
+ *
+ * @ViewsArgumentDefault(
+ *   id = "og_group_membership",
+ *   title = @Translation("Group memberships from current user")
+ * )
+ */
+class Membership extends ArgumentDefaultPluginBase implements CacheableDependencyInterface {
+
+  /**
+   * The OG context provider.
+   *
+   * @var \Drupal\Core\Plugin\Context\ContextProviderInterface
+   */
+  protected $ogContext;
+
+  /**
+   * Constructs a new User instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Plugin\Context\ContextProviderInterface $og_context
+   *   The OG context provider.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ContextProviderInterface $og_context) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->ogContext = $og_context;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('og.context')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function defineOptions() {
+    $options = parent::defineOptions();
+    $options['og_group_membership'] = ['default' => ''];
+
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
+    $form['og_group_membership'] = [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getArgument() {
+    // Currently restricted to node entities.
+    return implode(',', $this->getCurrentUserGroups('node'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheMaxAge() {
+    return Cache::PERMANENT;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    return ['og_role'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
+    $group = $this->getGroup();
+    if ($group instanceof ContentEntityInterface) {
+      $tag = $group->getEntityTypeId() . ':' . $group->id();
+      return Cache::buildTags('og-group-content', [$tag]);
+    }
+    return [];
+  }
+
+  /**
+   * Returns groups that current user is a member of.
+   *
+   * @return \Drupal\Core\Entity\ContentEntityInterface|null
+   *   The group, or NULL if no group is found.
+   */
+  protected function getCurrentUserGroups($entity_type = 'node') {
+    $user = \Drupal::currentUser();
+    $membership_manager = \Drupal::service('og.membership_manager');
+    $groups = $membership_manager->getUserGroupIds($user);
```

Comment:
`\Drupal::currentUser()` returns an `AccountProxyInterface`, the method needs `AccountInterface`, so.. `\Drupal::currentUser()->getAccount()`